system/include/emscripten/val.h

```diff
@@ -335,7 +337,13 @@ class val {
 
     template<typename T>
     static val array(const std::vector<T>& vec) {
-        return array(vec.begin(), vec.end());
+      if constexpr (std::is_arithmetic<T>::value && !std::is_same<T, bool>::value) {
```

Comment:
You can instead reuse `typeSupportsMemoryView` function that performs a bit more precise checks.